src/Microsoft.ML.Core/Data/IRowToRowMapper.cs

```diff
@@ -15,7 +16,15 @@ namespace Microsoft.ML.Data
     /// so to rebind, the same input column names must be used.
     /// Implementations of this interface are typically created over defined input <see cref="DataViewSchema"/>.
     /// </summary>
-    public interface IRowToRowMapper
+    public interface IRowToRowMapper : IRowToRowMapperBase
```

Comment:
>IRowToRowMapperBase [](start = 39, length = 19)

Can we please not introduce this interface? As far as I see, it exists so that we can have a common "descent" point for `IRowToRowMapper` and `ISchemaBoundRowMapper`. But the latter is internal and should remain so. I would therefore prefer that there be *no* relationship between the two. There is no reason for `IRowToRowMapperBase` to have a relationship with `ISchemaBoundRowMapper` as there is for it to have a relationship with `IValueMapper`. (BTW, in spirit, `ISchemaBoundRowMapper` is far closer to `IValueMapper`). So let's keep our inheritance structure clean. I do not want `IRowToRowMapper` to be complicated in this fashion.

Reply:
So: we'll have separate interfaces, and *not* pretend that they're somehow related, when in fact they really are not?

Reply:
I created the base interface, to reduce code redundancy. Without it, the ISchemaBoundRowMapper looks just like the IRowToRowMapper. 

Re-winding the change. 

---
In reply to: [260019470](https://github.com/dotnet/machinelearning/pull/2504#discussion_r260019470) [](ancestors = 260019470)

Reply:
It *looks* like it, but the objection of @yaeldekel is that it doesn't *act* like it, and that is deliberate and desirable, as far as I can tell. And if it impacts out public surface we have to hold it I think to a higher standard. Anyway, thanks for looking at this.

---
In reply to: [260026858](https://github.com/dotnet/machinelearning/pull/2504#discussion_r260026858) [](ancestors = 260026858,260019470)